### ISSUE TITLE
Correct documentation on the `editorconfig` option.

### ISSUE THREAD

**Issue by mangelozzi** [prettier/prettier#15255]
The issue is related to #6176

The documentation describes a `editorconfig` option, and trying to implement results in an error (tried both ways):
```
[warn] Ignored unknown option { editorconfig: true }.
[warn] Ignored unknown option { options: { editorconfig: true } }.
```
#6176 Says there is no `editorconfig` option, please update the documentation: 
<https://prettier.io/docs/en/configuration#editorconfig>
> If **options.editorconfig** is true and an [.editorconfig file](https://editorconfig.org/) is in your project, Prettier will parse it and convert its properties to the corresponding Prettier configuration. This configuration will be overridden by .prettierrc, etc.

**Comment by arkinmodi** [comment]
`editorconfig` is also listed as an option in the JSON schema. 

https://prettier.io/docs/en/configuration#configuration-schema
http://json.schemastore.org/prettierrc

```json
        "editorconfig": {
          "description": "Whether parse the .editorconfig file in your project and convert its properties to the corresponding Prettier configuration. This configuration will be overridden by .prettierrc, etc.",
          "default": false,
          "type": "boolean"
        },
```

**Comment by mhagnumdw** [comment]
Configuring Prettier with .editorconfig, I ran into the same problem. [Reading the doc](https://prettier.io/docs/en/configuration.html#editorconfig) and [checking the JSON schema](https://prettier.io/docs/en/configuration.html#configuration-schema), `editorconfig=true` should be a valid option. But when running `npx prettier . --check` get `[warn] Ignored unknown option { editorconfig: true }`.

version: "prettier": "3.0.3"

`.prettierrc.yaml` (I also tried with .prettierrc and .prettierrc.json)

```yaml
editorconfig: true # problem!
printWidth: 100
singleQuote: true
trailingComma: "es5"
semi: true
arrowParens: "avoid"
```

vscode
![image](https://github.com/prettier/prettier/assets/17688846/8e832998-742b-4115-a2e7-499e4b2bec79)

Where am I going wrong?

**Comment by kingyue737** [comment]
`editorconfig` in schema was added by me as I misunderstood the documentation https://github.com/SchemaStore/schemastore/pull/2355. I've reopened a PR to remove it as a maintainer https://github.com/prettier/prettier/issues/6176#issuecomment-498808239 confirmed that this option only exists in function API, not CLI (CLI will always read the `editorconfig` file if it exists). Sorry for bothering many other users😭😭😭.

**Comment by JoepKockelkorn** [comment]
Maybe we can mention in the docs ([here](https://prettier.io/docs/en/configuration#editorconfig)) that the CLI always reads the `.editorconfig` file. I've been here multiple times already 😆.

**Comment by shantanu-bbai** [comment]
> Maybe we can mention in the docs ([here](https://prettier.io/docs/en/configuration#editorconfig)) that the CLI always reads the `.editorconfig` file. I've been here multiple times already 😆.

I don't think this is always true. Or at least, the cli is not respecting the use of tabs unless it's specified in the .prettierrc file.

**Comment by JoepKockelkorn** [comment]
@shantanu-bbai I don't experience the behaviour you describe. These are my config files and I reliably get tabs as the indentation character:
### .prettierrc
```
{
	"singleQuote": true,
	"semi": false,
	"bracketSpacing": true,
	"bracketSameLine": false,
	"trailingComma": "all"
}
```
### .editorconfig
```
root = true

[*]
charset = utf-8
indent_style = tab
indent_size = 2
insert_final_newline = true
trim_trailing_whitespace = true
max_line_length = 80

[*.md]
max_line_length = off
trim_trailing_whitespace = false
```
Used version of prettier is `2.8.8`.

**Comment by shantanu-bbai** [comment]
Hi @JoepKockelkorn, I'm fairly new to prettier and csharpier (we're just starting to use both) so its entirely possible I don't know what I'm doing. My initial plan was not to use the .prettierrc file at all (though I had planned on having an empty file in our git project just so new devs would realize that we're using prettier) and just use the .editorconfig for everything.

Here are the contents of my files:

.prettierrc (what I ended up having to populate instead of having an empty file - these configurations are picked up by the prettier commadline execution)

`trailingComma: 'es5'
tabWidth: 4
semi: false
singleQuote: true
useTabs: true`

.editorconfig

`# prettier configuration
[*]
charset = utf-8
insert_final_newline = true
end_of_line = lf
indent_style = tabs
indent_size = 4
max_line_length = 150
use_tabs = true`

So maybe I should step back. I was looking through the project docs and I don't believe I've found a full set of possible configuration options listed. Most of the options I've found through stackoverflow threads. Could you point me to an exhaustive set of options for prettier?

Also, can you tell me what has to go in prettierrc (i.e. cannot be configured via .editorconfig) if anything?

Lastly, can you tell me what I'm doing wrong with my configs (editorconfig specifically)?

**Comment by JoepKockelkorn** [comment]
@shantanu-bbai I will offer my help to you, but I have no experience with csharpier. Besides, it's unclear what you're trying to achieve. What are you using these tools for, what's **your** goal? 

It's good to realise that editorconfig and prettier are both formatting tools but they slightly differ in scope and features. Editorconfig tries to bring a standardised formatting experience from the IDE perspective for nearly every file format. Without using an IDE (and plugin) to format your files, your `.editorconfig` settings won't do anything. It has no CLI to format files without using an IDE. [You can find a list of supported settings here](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties).

Prettier is similar, but different. It is much more opinionated, which means that there are some settings to tweak its behaviour, but some things just work as they do without a way to configure its behaviour. Also, it only works on certain file formats, not just every file. Prettier is pluggable though, so you can make it work for xml files using https://github.com/prettier/plugin-xml, but it doesn't work out of the box in contrast to editorconfig. Also, Prettier has extensions/plugins for your IDE to format files there, but it also has a CLI to format independent of your IDE. This opens possibilities to use it in CI to guarantee your files adhere to the configured prettier format using [`prettier . --check`](https://prettier.io/docs/en/cli#--check).

Some settings of Prettier overlap with editorconfig settings. Try searching for 'editorconfig' on the [Prettier configuration page](https://prettier.io/docs/en/options). For those, it's better to omit them from the Prettier configuration so Prettier will read them from the `.editorconfig`.

The documentation websites of both tools are pretty good, so any remaining questions you have will probably be answered by reading them:
- [Prettier](https://prettier.io/docs/en/)
- [Editorconfig](https://editorconfig.org/)

**Comment by hallipr** [comment]
@shantanu-bbai, it looks like [Prettier's options documentation](https://prettier.io/docs/en/options.html) lists the `.editorconfig` settings that will be imported.

printWidth
> Setting max_line_length in an [.editorconfig file](https://editorconfig.org/) will configure Prettier’s print width, unless overridden.

tabWidth
> Setting indent_size or tab_width in an [.editorconfig file](https://editorconfig.org/) will configure Prettier’s tab width, unless overridden.

useTabs
> Setting indent_style in an [.editorconfig file](https://editorconfig.org/) will configure Prettier’s tab usage, unless overridden.

Maintainers, is this an exhaustive list of settings read from `.editorconfig`?

**Comment by JoepKockelkorn** [comment]
You've missed `end_of_line`.

**Comment by JoepKockelkorn** [comment]
> Docs still say "If options.editorconfig is true" and it's very confusing. If I understand this thread correctly, this option is not a real option?
> 
> [prettier.io/docs/en/configuration#editorconfig](https://prettier.io/docs/en/configuration#editorconfig)

I think it is an invalid option indeed ([except the opt-out flag on the cli](https://prettier.io/docs/en/cli#--no-editorconfig)), but it still survived in the documentation in multiple places somehow.
- https://prettier.io/docs/en/api#prettierresolveconfigfileurlorpath--options
- https://prettier.io/docs/en/configuration#editorconfig

**Comment by abiriadev** [comment]
I would like to know if Prettier supports `.editorconfig` out of the box.

If `options.editorconfig` is only accessible through the API and cannot be configured directly via the configuration file, what is the recommended way to integrate Prettier with an `.editorconfig` file?

**Comment by JoepKockelkorn** [comment]
As mentioned before in this issue, Prettier already reads the `.editorconfig` file by default. That means ([link](https://prettier.io/docs/en/configuration#editorconfig)):

> Prettier will parse it and convert its properties to the corresponding Prettier configuration.

Be aware though: if you omit any of `max_line_length`, `indent_size`, `indent_style` or `end_of_line` in the `.editorconfig` options, Prettier will use the default as mentioned on https://prettier.io/docs/en/options. This could be pretty frustrating for some options as they might not come from an `.editorconfig` but from a `.gitattributes` file.

Maybe [my blog post on Prettier + Editorconfig](https://joepkockelkorn.com/blog/combining-gitattributes-editorconfig-and-prettier) can also shed some light on the interoperability issues.

**Comment by catdadcode** [comment]
I created PR #16335 to remove this wording from the docs and more clearly describe that `.editorconfig` will just work out of the box for the most part.

**Comment by abiriadev** [comment]
> As mentioned before in this issue, Prettier already reads the `.editorconfig` file by default. That means ([link](https://prettier.io/docs/en/configuration#editorconfig)):

Thank you for the clarification! I can't believe I missed this in the documentation.

**Comment by talatkuyuk** [comment]
> Be aware though: if you omit any of `max_line_length`, `indent_size`, `indent_style` or `end_of_line` in the `.editorconfig` options, Prettier will use the default as mentioned on https://prettier.io/docs/en/options. This could be pretty frustrating for some options as they might not come from an `.editorconfig` but from a `.gitattributes` file.

Exactly, missing `max_line_length` in `.editorconfig` caused prettier doesn't work on `objectWrap` weirdly, (actually, I set `printWidth` is 96 in `.prettierrc.json`). When I put `max_line_length = 96` in `.editorconfig`, then prettier acted correctly.

**Comment by gshpychka** [comment]
Open a PR to remove the lat mention of the option from the docs: https://github.com/prettier/prettier/pull/17168

**Comment by dev-nicolaos** [comment]
I believe this issue can be closed since:

- `options.editorconfig` is documented for the API (https://prettier.io/docs/api#prettierresolveconfigfileurlorpath--options)
- there is (correctly) no mention of a non-existent`editorconfig` option on the CLI documentation page
